### DISCUSSION TITLE
How do I use custom motions?

### DISCUSSION THREAD

**Discussion by AstrickHarren** [kylechui/nvim-surround#303] [Q&A] [answered]
I have the following maps: 
```lua 
		wk.register({
			mode = { "n", "x", "o" },
			g = {
				name = "+Goto",
				["*"] = "Search forward under cursor (exprwise)",
				["#"] = "Search backward under cursor (exprwise)",
				h = { "0", "First character of line" },
				s = { "^", "First non-blank character of line" },
				l = { "g_", "Last character of line" },
				e = { "G", "Last non-blank character of file" },
			},
		})
```
Apologize for not writing in vim api format, but you get the idea. Now, I use `ge` to go to the end of file (Kakonue mapping), and with the above keymapping, I am able to copy the text by `yge` but it seems `ysge` doesn't work for surroundings. As a matter of fact, `ysge` seems to stick to the original definition of `ge`, which is `go to the prev end of word`. 

So how can I ensure `ysge` surround the rest of the file?

**Comment by kylechui** [comment, marked as answer]
I'm not 100% sure how which-key is defining these mappings. If you just do `vim.keymap.set("o", "ge", "G")`, it should work.
